app/components/primer/button_component.rb

```diff
@@ -28,12 +28,20 @@ class ButtonComponent < Primer::Component
     # Icon to be rendered in the button.
     #
     # @param system_arguments [Hash] Same arguments as <%= link_to_component(Primer::OcticonComponent) %>.
-    renders_one :icon, Primer::OcticonComponent
+    renders_one :icon, lambda { |**system_arguments|
```

Comment:
I have some concerns with using utility classes to address the button spacing issues, though I'm happy to see it being addressed! I would prefer that we make layout changes in Primer CSS so there's consistency between PVC and Primer CSS. 

The React team is also drafting a new Button API that I've been reviewing/pairing with @pksjce on. I've started drafting an API change for Primer CSS, which would address this spacing issue but also introduce a breaking change. I don't think we should wait for those changes necessarily, so maybe its worth making this change in CSS and cutting a minor release in the meantime. What do you think @manuelpuyol ?

Reply:
Yeah, I'm a little concerned about this as well. Shouldn't spacing be handled by the caller?

Reply:
I _think_ it's OK to have a wrapping component handle spacing amongst its children. This prevents us from having to write something like `.btn .octicon { margin-right...`.

Reply:
I also think it's ok for the parent component to set the spacing of its children. The user won't be able to customize those spaces and we won't need CSS changes to achieve what we want

Reply:
Ah sorry I didn't realize the spacing was on a child component, carry on!

Reply:
@joelhawksley I think we _should_ handle it with CSS, and `btn` should have specific styles for handling `icon` or any kind of visual. Not by targeting `.octicon` but by providing a specific class like `.btn-icon` or even more generic `.leadingVisual` as we've introduced in ActionList. I'm not a fan of using utility classes inside components, even if its not visible to the end user of PVC.

`button` should be addressed on a larger scale and the CSS should address all child element scenarios. That said, when we do work on these changes we can simply remove this utility assuming its not required anymore.

Reply:
I agree with @langermank. Placing the inner component bits together in CSS like we did with `actionList` makes it easier for us to build, test, and maintain these components long-term. 

Not all responsive adaptations we're using now can be done by utilities anyway, and it feels it'd be harder to keep only _some_ properties in a PVC file.

🙇

Reply:
We can aim to have these changes in PrimerCSS, but I think we can keep the utilities here while they aren't upstreamed :)